lizmap/modules/lizmap/controllers/edition.classic.php

```diff
@@ -424,7 +425,7 @@ private function getDataFields($datasource){
         // If postgresql, get real geometryType from geometry_columns (jelix prop gives 'geometry')
         if( $this->provider == 'postgres' and $this->geometryType == 'geometry' ){
           $cnx = jDb::getConnection($this->layerId);
-          $res = $cnx->query('SELECT type FROM geometry_columns WHERE f_table_name = '.$cnx->quote($this->tableName));
+          $res = $cnx->query('SELECT type FROM geometry_columns WHERE f_table_schema = '.$cnx->quote($this->schemaName).' AND f_table_name = '.$cnx->quote($this->tableName));
```

Comment:
The better way to code it could be:
```php
        $sql = "SELECT type FROM geometry_columns";
        $sql.= " WHERE 2>1";
        $sql.= " AND (f_table_schema = ANY (current_schemas(false)) OR t_table_schema = " . $cnx->quote($schema) . ")";
        $sql.= " AND f_table_name=".$cnx->quote($tableAlone);
        $res = $cnx->query($sql);
```

Reply:
the test on the schema name `f_table_schema = ANY (current_schemas(false))` returns two rows because two tables shared the same name in two schema. This will cause the same error.
This works fine in my project : 
```php
 $sql = "SELECT type FROM geometry_columns";
        $sql.= " WHERE 2>1";
        $sql.= " AND t_table_schema = " . $cnx->quote($schema);
        $sql.= " AND f_table_name=".$cnx->quote($tableAlone);
        $res = $cnx->query($sql);
```

Reply:
why the `2>1` test ?